docs/docs/Tutorials/ROS2/CPPTemplate.md

```diff
@@ -0,0 +1,394 @@
+cmake_minimum_required(VERSION 3.8)
+project(mbari_wec_template_cpp)  # Update ${PROJECT_NAME}
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+# find dependencies
+find_package(ament_cmake REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(buoy_interfaces REQUIRED)
+find_package(buoy_api_cpp REQUIRED COMPONENTS buoy_api)
+
+add_executable(${PROJECT_NAME} src/controller.cpp)
+target_link_libraries(${PROJECT_NAME} PUBLIC buoy_api_cpp::buoy_api)
+ament_target_dependencies(${PROJECT_NAME} PUBLIC rclcpp buoy_interfaces)
+target_include_directories(${PROJECT_NAME} PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:include>)
+target_compile_features(${PROJECT_NAME} PUBLIC c_std_99 cxx_std_17)  # Require C99 and C++17
+```
+
+- `launch/controller.launch.py` (line 22)
+
+``` py linenums="22" hl_lines="1" title="launch/controller.launch.py"
+package_name = 'your_package_name'   # Update package name (same as in CMakeLists.txt)
+
+def generate_launch_description():
+    ld = LaunchDescription()
+    config = os.path.join(
+        get_package_share_directory(package_name),
+        'config',
+        'controller.yaml'
+        )
+
+    node = Node(
+        package=package_name,
+        name=package_name,  # ensure same as name in config.yaml
+        executable=package_name,
+```
+
+- `config/controller.yaml` (line 1)  
+    Update first line with your controller name (same as node name in launch file)
+
+``` yaml linenums="1" hl_lines="1" title="config/controller.yaml"
+/your_controller_name:
+  ros__parameters:
+    foo: 1.0
+```
+
+and rename the folder
+
+- `include/mbari_wec_template_cpp` (containing `controller.hpp` and `control_policy.hpp`) to your
+  package name
+
+resulting in the following folder structure:
+
+```
+<your_package_name>
+    ├── CMakeLists.txt
+    ├── config
+    │   └── controller.yaml
+    ├── include
+    │   └── <your_package_name>
+    │       ├── controller.hpp
+    │       └── control_policy.hpp
+    ├── launch
+    │   └── controller.launch.py
+    ├── LICENSE
+    ├── package.xml
+    ├── README.md
+    └── src
+        └── controller.cpp
+```
+
+Update the include paths in
+
+- `controller.cpp` (lines 15-16)
+``` cpp linenums="15" hl_lines="1 2" title="src/controller.cpp"
+#include <mbari_wec_template_cpp/control_policy.hpp>  // update include path
+#include <mbari_wec_template_cpp/controller.hpp>  // update include path
+```
+
+- `control_policy.hpp` (line 15)
+``` cpp linenums="15" hl_lines="1" title="include/your_package_name/control_policy.hpp"
+#include <mbari_wec_template_cpp/controller.hpp>  // update include path
+```
+
+Update include guards in
+
+- `controller.hpp`
+``` cpp linenums="15" hl_lines="1 2" title="include/your_package_name/controller.hpp"
+#ifndef YOUR_PACKAGE_NAME__CONTROLLER_HPP_
+#define YOUR_PACKAGE_NAME__CONTROLLER_HPP_
+```
+`...`
+``` cpp linenums="74" hl_lines="1"
```

Comment:
Is it intentional that this box is gray, and the box above is white? I guess the gray is the style when you don't put a `title` in it? That's fine here.

Reply:
Yeah, it looks lame to me too... I wanted to just leave the middle section of code out but I couldn't find a way to do a break in line numbers in a single code block